src/inputs/internal/SelectFieldBase.tsx

```diff
@@ -61,19 +63,36 @@ export function SelectFieldBase<O, V extends Value>(props: BeamSelectFieldBasePr
     disabled,
     readOnly,
     onSelect,
-    options: maybeOptions,
+    options,
     multiselect = false,
-    getOptionLabel,
-    getOptionValue,
-    getOptionMenuLabel = getOptionLabel,
+    getOptionLabel: gol,
+    getOptionValue: gov,
+    getOptionMenuLabel: goml,
     values = [],
     nothingSelectedText = "",
     contrast,
     disabledOptions,
     borderless,
+    unsetLabel,
     ...otherProps
   } = props;
 
+  // Call `initializeOptions` to prepend the `unset` option if the `unsetLabel` was provided.
+  const maybeOptions = useMemo(() => initializeOptions(options, unsetLabel), [options, unsetLabel]);
+  // Memoize the callback functions and handle the `unset` option if provided.
+  const getOptionLabel = useCallback(
+    (o: O) => (unsetLabel && isUnsetOption(o) ? unsetLabel : gol(o)),
+    [gol, unsetLabel],
+  );
+  const getOptionValue = useCallback(
+    (o: O) => (unsetLabel && isUnsetOption(o) ? (o.id as V) : gov(o)),
```

Comment:
Suggestion: The `o.id as V` could probably just be `undefined`?

Reply:
We'd need to do `undefined as V`, I believe. Otherwise we'd get issues like "Type 'undefined' is not assignable to type 'V'."

Reply:
This rewrite ended up not being necessary as we already handled values set to `undefined`.